### PR TITLE
Move EIP-712 crate back to parity-ethereum

### PR DESCRIPTION
**Pull request by seunlanlege**
converted to rust 2018 module, added readme, license

### CODE REVIEW

File: util/EIP-712/src/lib.rs [hunk 1]
```diff
@@ -156,35 +156,22 @@
 //! }
 //! ```
 
-#![warn(missing_docs, unused_extern_crates)]
+#![warn(missing_docs)]
```
Comment (tomusdrw) [inline util/EIP-712/src/lib.rs]: Why remove `unused_extern_crates`?

Comment (seunlanlege) [reply util/EIP-712/src/lib.rs]: this module uses rust 2018, no need for extern crates

File: util/EIP-712/Cargo.toml [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
-name = "eip712"
+name = "eip-712"
 version = "0.1.0"
 authors = ["Parity Technologies <admin@parity.io>"]
+repository = "https://github.com/paritytech/eip-712"
+documentation = "https://docs.rs/eip712"
+readme = "README.md"
+description = "eip-712 encoding"
+keywords = ["eip-712", "eip712", "eip"]
+license = "MIT"
```
Comment (tomusdrw) [inline util/EIP-712/Cargo.toml]: should be GPL

File: util/EIP-712/LICENSE [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Copyright 2015-2017 Parity Technologies
```
Comment (tomusdrw) [inline util/EIP-712/LICENSE]: This file should include a copy of `GPL` license afaict

File: util/EIP-712/Cargo.toml [hunk 4]
```diff
@@ -1,7 +1,14 @@
 [package]
-name = "eip712"
+name = "eip-712"
 version = "0.1.0"
 authors = ["Parity Technologies <admin@parity.io>"]
+repository = "https://github.com/paritytech/eip-712"
```
Comment (tomusdrw) [inline util/EIP-712/Cargo.toml]: please update the repo

File: util/EIP-712/Cargo.toml [hunk 5]
```diff
@@ -1,7 +1,14 @@
...
+documentation = "https://docs.rs/eip712"
```
Comment (tomusdrw) [inline util/EIP-712/Cargo.toml]: this is missing dash now it seems

**Pull request merged**